hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNetworkContext.java

```diff
@@ -353,16 +354,29 @@ public long lastScannedEntity() {
 	}
 
 	public void resetAutoRenewSummaryCounts() {
+		if (isImmutable()) {
+			throw new IllegalStateException("Cannot reset auto-renew summary counts, context is immutable!");
+		}
```

Comment:
Should extract this code to a separate function (that receives a string as the message for throwing an ISS) for reuse.